### PR TITLE
Comma srl to pipeline 

### PR DESCRIPTION
**Pull request by danyaljj**
Adds Comma-SRL. History kept (big files removed). 
This also moves stanford parser handlers outside the pipeline into their own module to get rid of circularity in the project structure.

### CODE REVIEW

**Comment by danyaljj** [comment]
On halt: there is a circularity here.

**Comment by danyaljj** [comment]
@mssammon @bhargav ready to be reviewed.

**Comment by danyaljj** [comment]
> Main concern is the Stanford version. Pom has 3.3.1, while some portions of code has 3.1.1.

Good catch! Will fix it. 

 > This PR also includes dataset corpus under commasrl/data/corpus/* - Around 6k lines. And some more annotation related files. Just checking if adding them was intentional.

Yeah I kept it intentional. Text data, don't think it's gonna take much space. I can drop it.  

 > Some utility methods can be moved into CoreUtils (esp. PrettyPrintParseTree). Maybe in a later PR.

Good point; for future.

**Comment by danyaljj** [comment]
Fixed Stanford version mess and dropped the redundant data. 
For future PR, I'll make sure the data is put somewhere accessible for future users, if they want to retrain it. (probably on group website)

File: commasrl/pom.xml [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+                <version>2.12</version>
+            </extension>
+        </extensions>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.0</version>
+                <configuration>
+                    <source>1.7</source>
```
Comment (mssammon) [inline commasrl/pom.xml]: main cogcomp-nlp uses 1.8 -- any reason for 1.7 here?

Comment (danyaljj) [inline commasrl/pom.xml]: I'll update it. 
Is that the only comment?

**Comment by mssammon** [comment]
Probably not. Review in progress.

File: commasrl/src/main/java/edu/illinois/cs/cogcomp/comma/datastructures/Sentence.java [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+import edu.illinois.cs.cogcomp.core.utilities.StringUtils;
+
+/**
+ * Used to group all the commas in a sentence. The commas can then be accessed based on positional
+ * relations to each other within the sentence
+ * 
+ * @author navari
+ *
+ */
+public class Sentence implements Serializable {
```
Comment (mssammon) [inline commasrl/src/main/java/edu/illinois/cs/cogcomp/comma/datastructures/Sentence.java]: This has the same name as a core-utilities class -- should we change it to make it clear where this is used (without checking imports)?

File: commasrl/pom.xml (hunk 1 above)
Comment (danyaljj) [inline commasrl/pom.xml]: Fixed.

File: commasrl/src/main/java/edu/illinois/cs/cogcomp/comma/annotators/PreProcessor.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+import edu.illinois.cs.cogcomp.core.datastructures.textannotation.TextAnnotation;
+import edu.illinois.cs.cogcomp.core.utilities.configuration.Configurator;
+import edu.illinois.cs.cogcomp.core.utilities.configuration.ResourceManager;
+import edu.illinois.cs.cogcomp.curator.CuratorConfigurator;
+import edu.illinois.cs.cogcomp.curator.CuratorFactory;
+import edu.illinois.cs.cogcomp.ner.NERAnnotator;
+import edu.illinois.cs.cogcomp.ner.NerAnnotatorManager;
+import edu.illinois.cs.cogcomp.nlp.tokenizer.IllinoisTokenizer;
+import edu.illinois.cs.cogcomp.nlp.tokenizer.Tokenizer;
+import edu.illinois.cs.cogcomp.pipeline.common.Stanford331Configurtor;
```
Comment (danyaljj) [inline commasrl/src/main/java/edu/illinois/cs/cogcomp/comma/annotators/PreProcessor.java]: fixed.

File: commasrl/src/main/java/edu/illinois/cs/cogcomp/comma/bayraktar/BayraktarPatternLabeler.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    static {
+        CommaProperties properties = CommaProperties.getInstance();
+        String[] labels =
+                {"Attribute", "Complementary", "Interrupter", "Introductory", "List", "Quotation",
+                        "Substitute", "Locative"};
+        BAYRAKTAR_PATTERN_TO_COMMA_LABEL = new HashMap<>();
+        String ANNOTATION_SOURCE_DIR = properties.getBayraktarAnnotationsDir() + File.separator;
+        if (properties.useDatastoreToReadData()) {
+            try {
+                Datastore ds = new Datastore("http://smaug.cs.illinois.edu:8080");
```
Comment (danyaljj) [inline commasrl/src/main/java/edu/illinois/cs/cogcomp/comma/bayraktar/BayraktarPatternLabeler.java]: Good point. Fixed.

File: commasrl/src/main/java/edu/illinois/cs/cogcomp/comma/datastructures/Sentence.java (hunk 2 above)
Comment (danyaljj) [inline commasrl/src/main/java/edu/illinois/cs/cogcomp/comma/datastructures/Sentence.java]: Good point. Fixed it.

**Pull request merged**